service/qos/service_level_controller.hh

```diff
@@ -67,6 +68,23 @@ public:
         virtual future<> drop_service_level(sstring service_level_name) const = 0;
     };
     using service_level_distributed_data_accessor_ptr = ::shared_ptr<service_level_distributed_data_accessor>;
+
+    class subscriber_registry {
+    private:
+        qos_configuration_change_subscriber* _qccs;
+        friend class service_level_controller;
```

Comment:
I still think that this whole wrapper class is unnecessary, but if it isn't then I have another question - is this friendship used anywhere? I don't see where.

Reply:
It is used in the parallel for each in the `service_level_controller::notify_*` functions but I am going to remove this so never mind :slightly_smiling_face:

Reply:
removed in the new version